boards/FSM/Src/Io/Io_SteeringAngleSensor.c

```diff
@@ -1,11 +1,19 @@
+#include <math.h>
 #include "Io_Adc.h"
 #include "Io_SteeringAngleSensor.h"
 
+#define MIN_STEERING_VOLTAGE (0.0f)
+#define MAX_STEERING_VOLTAGE (1.0f)
+
 float Io_SteeringAngleSensor_GetAngleDegree(void)
 {
     // Steering angle = (ADC Voltage - Voltage Offset) * Degree Per Volt
 
     const float STEERING_ANGLE_VOLTAGE_OFFSET = 1.85f;
     const float DEGREE_PER_VOLT               = 360.0f / 3.3f;
-    return DEGREE_PER_VOLT * (Io_Adc_GetChannel1Voltage() - STEERING_ANGLE_VOLTAGE_OFFSET);
+    float steering_voltage = Io_Adc_GetChannel1Voltage();
+    if(!(MIN_STEERING_VOLTAGE <= steering_voltage && steering_voltage <= MAX_STEERING_VOLTAGE)){
+        return NAN;
+    }
+    return DEGREE_PER_VOLT * (steering_voltage - STEERING_ANGLE_VOLTAGE_OFFSET);
```

Comment:
Using NaNs seems sketchy - can we use inrangecheck instead here? Generally I don't like the idea of using NaNs

Reply:
oh yeah true, however @joethrstn was saying something about NAN used as open/short circuit while inRangeCheck is for plausability?

Reply:
In my opinion, seems like a bad practice to use NaNs to indicate errors. Its not obvious when looking at the function signature that it could return NaN, and will cause errors if not caught. 

I'd suggest InRangeCheck (since I think all the failure modes here are out of range errors?) or a seperate function that checks specifically if brake is OC or SC (which I think is how it used to work anyway?)

Reply:
Fixed with f3f4a15